services/catalog/src/simcore_service_catalog/services/services_api.py

```diff
@@ -267,17 +274,17 @@ async def update_service(
         await repo.upsert_service_access_rights(new_access_rights)
 
         # then delete the ones that were removed
-        remove_gids = [gid for gid in before_gids if gid not in update.access_rights]
-        delete_access_rights = [
+        rm_access_rights = [
             ServiceAccessRightsAtDB(
                 key=service_key,
                 version=service_version,
                 gid=gid,
                 product_name=product_name,
             )
-            for gid in remove_gids
+            for gid in previous_gids
+            if gid not in update.access_rights
         ]
-        await repo.delete_service_access_rights(delete_access_rights)
+        await repo.delete_service_access_rights(rm_access_rights)
```

Comment:
```suggestion
        await repo.delete_service_access_rights(removed_access_rights)
```
Please let's not spare 5 characters...